app/src/main/java/org/mozilla/fenix/widget/VoiceSearchActivity.kt

```diff
@@ -37,6 +41,23 @@ class VoiceSearchActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
```

Comment:
I guess we should use the extension here as well. 

Here the receiver expects the callback to be called even without `RESULT_OK`. But the extension function consumes `results` that are not `RESULT_OK` without any feedback, which might be unexpected. Maybe adding a second callback for cancelled state for edge cases?

Reply:
I think we should just use this here since currently there is only one activity doing this.  Once there are more than one activity, moving it to an extension function makes sense then.